### PR TITLE
announce about progress on section progress page

### PR DESCRIPTION
**Pull request by Bjvanminnen**
image

Show a message about progress on courses in which it's disabled. If you change courses in the dropdown the message will hide as appropriate.

### CODE REVIEW

File: pegasus/helpers/section_api_helpers.rb [hunk 1]
```diff
@@ -252,6 +252,14 @@ def self.valid_courses(user_id = nil)
         end
   end
 
+  def self.progress_disabled_courses(user_id = nil)
+    disabled_courses = valid_courses(user_id).select do |course|
+      script_name = Script.find(course[:id]).name
+      !Gatekeeper.allows('postMilestone', where: {script_name: script_name}, default: true)
+    end
```
Comment (Bjvanminnen) [inline pegasus/helpers/section_api_helpers.rb]: We need to be able to know whether to show this message or not on a per course basis. This means we need to get the gatekeeper settings for each course on page load. Not sure if this ends up being more querying than we'd like.

Comment (Bjvanminnen) [reply pegasus/helpers/section_api_helpers.rb]: @ashercodeorg seems like someone who might have opinions on this

File: pegasus/helpers/section_api_helpers.rb [hunk 2]
```diff
@@ -252,6 +252,14 @@ def self.valid_courses(user_id = nil)
         end
   end
 
+  def self.progress_disabled_courses(user_id = nil)
+    disabled_courses = valid_courses(user_id).select do |course|
+      script_name = Script.find(course[:id]).name
```
Comment (ashercodeorg) [inline pegasus/helpers/section_api_helpers.rb]: Any reason not to use the script cache (via `Script.get_from_cache(course[:id]).name`)?

Comment (Bjvanminnen) [reply pegasus/helpers/section_api_helpers.rb]: nope. will fix

File: pegasus/helpers/section_api_helpers.rb (hunk 1 above)
Comment (ashercodeorg) [reply pegasus/helpers/section_api_helpers.rb]: AFAIK, querying the `Gatekeeper` is cheap. Perhaps @wjordan can say for certain.

File: pegasus/helpers/section_api_helpers.rb [hunk 3]
```diff
@@ -252,6 +252,14 @@ def self.valid_courses(user_id = nil)
         end
   end
 
+  def self.progress_disabled_courses(user_id = nil)
```
Comment (ashercodeorg) [inline pegasus/helpers/section_api_helpers.rb:257]: Please document the method: what does it do, what does it return, etc.

**Comment by ashercodeorg** [comment]
LGTM, though you should get someone else to review the non-ruby code.

**Comment by Bjvanminnen** [comment]
Script.\* doesn't work on pegasus.

**Pull request merged**